framework/py/flwr/client/grpc_adapter_client/connection.py

```diff
@@ -45,7 +46,7 @@ def grpc_adapter(  # pylint: disable=R0913,too-many-positional-arguments
     tuple[
         int,
         Callable[[], tuple[Message, ObjectTree] | None],
-        Callable[[Message, ObjectTree], set[str]],
+        Callable[[MessageSendContext, ObjectTree], set[str]],
```

Comment:
Just curious, why not just add a new argument?

```suggestion
        Callable[[Message, ObjectTree, float], set[str]],
```